dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/WorkFlowLineageMapper.xml

```diff
@@ -161,4 +161,58 @@
         </if>
         AND a.task_type = 'DEPENDENT'
     </select>
+
+    <select id="queryTaskDepOnProcess" resultType="org.apache.dolphinscheduler.dao.entity.TaskMainInfo">
+        select td.id
+            , td.name as taskName
+            , td.code as taskCode
+            , td.version as taskVersion
+            , td.task_type as taskType
+            , ptr.process_definition_code as processDefinitionCode
+            , pd.name as processDefinitionName
+            , pd.version as processDefinitionVersion
+            , pd.release_state as processReleaseState
+        from t_ds_task_definition td
+        join t_ds_process_task_relation ptr on ptr.post_task_code = td.code and td.version = ptr.post_task_version
+        join t_ds_process_definition pd on pd.code = ptr.process_definition_code and pd.version = ptr.process_definition_version
+        <where>
+            <if test="projectCode != 0">
+                and ptr.project_code = #{projectCode}
+            </if>
+            <if test="processDefinitionCode != 0">
+                and ptr.process_definition_code != #{processDefinitionCode}
+                and (
+                    td.task_params like concat('%"definitionCode":', #{processDefinitionCode}, '%')
+                    or td.task_params like concat('%"processDefinitionCode":', #{processDefinitionCode}, '%')
+                )
+            </if>
+        </where>
+    </select>
+
+    <select id="queryTaskDepOnTask" resultType="org.apache.dolphinscheduler.dao.entity.TaskMainInfo">
+        select td.id
+             , td.name as taskName
+             , td.code as taskCode
+             , td.version as taskVersion
+             , td.task_type as taskType
+             , ptr.process_definition_code as processDefinitionCode
+             , pd.name as processDefinitionName
+             , pd.version as processDefinitionVersion
+             , pd.release_state as processReleaseState
+        from t_ds_task_definition td
+        join t_ds_process_task_relation ptr on ptr.post_task_code = td.code and td.version = ptr.post_task_version
+        join t_ds_process_definition pd on pd.code = ptr.process_definition_code and pd.version = ptr.process_definition_version
+        <where>
```

Comment:
Here should add a '1=1' here, otherwise sql will report an error during execution.

Reply:
@WangJPLeo  I think mybatis will automatichandle the `and` or `or` keyword https://mybatis.org/mybatis-3/dynamic-sql.html#trim-where-set

Reply:
ok thx I'll take a look.